Source/NSubstitute.Acceptance.Specs/ReturnsForAll.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using NSubstitute.Acceptance.Specs.Infrastructure;
+using NSubstitute.Extensions;
+using NUnit.Framework;
+
+namespace NSubstitute.Acceptance.Specs
+{
+    [TestFixture]
+    public class ReturnsForAll
+    {
+        private IFluentSomething _fluentSomething;
+        private ISomething _something;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _fluentSomething = Substitute.For<IFluentSomething>();
+            _something = Substitute.For<ISomething>();
+            _fluentSomething.ReturnsForAll<IFluentSomething>(_fluentSomething);
+            _fluentSomething.ReturnsForAll<ISomething>(_something);
+        }
+
+        [Test]
+        public void Return_self_for_single_call()
+        {
+            Assert.That(_fluentSomething.Chain(), Is.SameAs(_fluentSomething));
+        }
+
+        [Test]
+        public void Return_self_for_chained_calls()
+        {
+           Assert.That(_fluentSomething.Chain().Me().Together(), Is.SameAs(_fluentSomething));
+        }
+
+        [Test]
+        public void Return_value_that_is_not_chainable()
+        {
+            Assert.That(_fluentSomething.SorryNoChainingHere(), Is.SameAs(_something));
+        }
+
+        [Test]
+        public void Return_samce_thing_for_multiple_calls()
+        {
+            var _first = _fluentSomething.SorryNoChainingHere();
```

Comment:
Sorry, my fault. Misprint in test name `Return_samce_thing_for_multiple_calls` in **samce**. The convention is to name local variables in camelCase, so instead of `var _first = ...` do `var first = ...`